app/scripts/modules/core/src/managed/ToggleManagedResourceForApplication.tsx

```diff
@@ -0,0 +1,130 @@
+        Careful! You’re about to stop Spinnaker from managing all resources in your application.
+      </span>
+      This feature should only be used if management is not working properly and manual intervention is required.{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery" target="_blank">
+        Check our documentation for more information
+      </a>
+      .
+    </p>
+    <p>
+      Do you just need to rollback?{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery/pinning/" target="_blank">
+        Please use the “pin” feature instead
+      </a>
+      .
+    </p>
+  </>
+);
+
+export const MDPausedStateDescription = () => {
+  return (
+    <p>
+      You’re about to resume management of cloud resources for your application. The latest good version approved for
+      deployment will be deployed to each environment.
+    </p>
+  );
+};
+
+export const showToggleManagedResourceModal = (props: IToggleManagedResourceForApplicationModalProps) =>
+  showModal(ToggleManagedResourceForApplicationModal, props, { maxWidth: 800 });
+
+export const ToggleManagedResourceForApplicationModal = memo(
+  ({ application, closeModal, dismissModal }: IToggleManagedResourceForApplicationModalProps) => {
+    const [isSubmitting, setSubmitting] = useState<boolean>(false);
+    const [actionError, setActionError] = useState<{ error: string; message: string } | null>(null);
+    const dataSource = application.getDataSource('environments');
+    let actionToBeTaken = 'Disable';
+    let description = <MDRunningStateDescription />;
+    let icon: IllustrationName = 'disableManagement';
+
+    if (application.isManagementPaused) {
+      actionToBeTaken = 'Resume';
+      description = <MDPausedStateDescription />;
+      icon = 'runManagement';
+    }
+
+    const handleActionInitiation = () => {
+      setActionError(null);
+      setSubmitting(true);
+      logClick(`${actionToBeTaken} Management`, application.name);
+      const call = application.isManagementPaused
+        ? ManagedWriter.resumeApplicationManagement(application.name)
+        : ManagedWriter.pauseApplicationManagement(application.name);
+      call
+        .then(() => dataSource.refresh(true).catch(() => null))
+        .then(() => {
+          application.isManagementPaused = actionToBeTaken === 'Disable';
+          dataSource.dataUpdated();
+          closeModal();
+        })
+        .catch((error: Error) => {
+          setActionError(error.data);
+          setSubmitting(false);
+        });
+    };
+    return (
+      <>
+        <ModalHeader>{actionToBeTaken} Managed Delivery</ModalHeader>
+        <ModalBody>
+          <div className="flex-container-h sp-padding-xl-yaxis">
+            <div style={{ minWidth: 145 }}>
+              <Illustration name={icon} />
+            </div>
+            <div className="sp-padding-xl">
+              <div>{description}</div>
+              <div>
+                {actionError &&
```

Comment:
Not saying it's better or worse than this, but FWIW I have been using `ValidationMessage` in other MD modal flows ofr handling submit errors: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/managed/PinArtifactModal.tsx#L157-L170